scalameta/contrib/src/main/scala/scala/meta/contrib/DocToken.scala

```diff
@@ -0,0 +1,275 @@
+package scala.meta.contrib
+
+import fastparse.all._
+import fastparse.core.Parsed
+
+/**
+  * Represents a scaladoc line.
+  */
+case class DocToken(kind: DocToken.Kind, name: Option[String], body: Option[String]) {
+
+  override def toString: String = {
+    ((name, body) match {
+      case (Some(n), _) => s"$kind(name=$n, body=${body.getOrElse("")})"
+      case (None, Some(kindBody)) => s"$kind($kindBody)"
+      case _ => kind.toString
+    }).replaceAll("\n", " ")
+  }
+
+  /**
+    * Obtains all the references from a [[DocToken]].
+    */
+  def references: Seq[DocToken.Reference] =
+    body
+      .map { b =>
+        {
+          def parseBodyFrom(idx: Int): Seq[DocToken.Reference] = {
+            DocToken.referenceParser
+              .parse(b, idx) match {
+              case p: Parsed.Success[String, _, _] =>
+                Seq(DocToken.Reference(p.value)) ++ parseBodyFrom(p.index)
+              case _ => Seq[DocToken.Reference]()
+            }
+          }
+
+          parseBodyFrom(0)
+        }
+      }
+      .getOrElse(Nil)
+}
+
+/**
+  * Companion object containing the classes required for describing an ScalaDoc token.
+  *
+  * The available tokens and their documentation are obtained from:
+  * @see http://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html
+  */
+object DocToken {
+
+  /**
+    * Parser that for obtaining a class reference like '[[scala.Some]]' from an scaladoc body.
+    */
+  private val referenceParser = P(
+    // Removes the elements previous to the references.
+    ((AnyChar ~ !"[[").rep ~ AnyChar).?
```

Comment:
It seems the regex approach is not cross-platform, it got a failing test while porting contrib to Scala.js. Good to know I can replace it with fastparse :)